## Section 1 - Imports

In [3]:
import pandas as pd # For Data Manipulation of Numerical Yables and Time Series 
import numpy as np # For Mathematical operations on arrays
import sklearn
import warnings
import matplotlib as plt 

## Section 2 - Data loading

In [4]:
patient = pd.read_csv('data/patient.csv', encoding='utf-8', engine='python')
clinical_codes = pd.read_csv('data/clinical_codes.csv', encoding='utf-8', engine='python')
medication = pd.read_csv('data/medication/medication.csv', encoding='utf-8', engine='python')
observation = pd.read_csv('data/observation/observation.csv', encoding='utf-8', engine='python')


## Section 2 ai- Visualise Top of Patient table

In [9]:
patient.drop(columns=["Unnamed: 0"], inplace=True, errors='ignore')
patient.head()


age  postcode patient_surname patient_givenname        date_of_birth  \
0    0   HX2 8WG            Bibi           Maureen  2023-02-11 00:00:00   
1    0  LS10 8MH         O'Brien             Karen  2022-11-08 00:00:00   
2    0  WF17 2XU          Turner              Mark  2023-01-28 00:00:00   
3    0  WF13 3RJ             Law         Frederick  2022-12-05 00:00:00   
4    0  LS99 9ZZ          Connor           Emerald  2022-11-17 00:00:00   

  date_of_death                     registration_guid  patient_id  \
0           NaN  9A70C9CD-D0F7-43F3-A05C-34C8F192E00C       10457   
1           NaN  57E3019C-25E6-4591-AA6D-3139AF37DC3F       11488   
2           NaN  A1700F02-134E-42FF-B537-1B7DA00F8C25        9844   
3           NaN  CEDE39B9-9261-4B21-9E07-92D4308E87D5        9972   
4           NaN  9532DEBE-42CC-40E4-820A-55DF603E0D23       18955   

          gender  
0         Female  
1         Female  
2           Male  
3           Male  
4  Indeterminate

## Section 2 aii - Visualise Bottom of Patient table

In [10]:
patient.tail()

age  postcode patient_surname patient_givenname        date_of_birth  \
4538    0  LS99 9ZZ           Lyons             Ahmir  2023-01-08 00:00:00   
4539    0  LS99 9ZZ      Williamson             Cyrus  2023-02-09 00:00:00   
4540    0  LS99 9ZZ           Healy            Kamari  2022-12-10 00:00:00   
4541    0  LS99 9ZZ          Bolton         Zachariah  2022-11-23 00:00:00   
4542    0  LS99 9ZZ         Johnson          Adrianna  2022-12-05 00:00:00   

     date_of_death                     registration_guid  patient_id  \
4538           NaN  61CE65F5-271B-4151-8700-70DE00CC0A9B       29589   
4539           NaN  4C16CFF4-520C-4B1C-8B03-9B214BDFAC76       19592   
4540           NaN  21107E5F-CE16-4669-9B6E-43671212E9D0       27505   
4541           NaN  9F8F0E53-7252-4998-A035-6809941C8A83       30629   
4542           NaN  F2E5C6F8-52B3-477B-83B2-955A9BB227A5       26947   

             gender  
4538         Female  
4539         Female  
4540  Indeterminate  
4541           Male  
4542         Female

In [11]:
patient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4543 entries, 0 to 4542
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   age                4543 non-null   int64 
 1   postcode           4438 non-null   object
 2   patient_surname    4543 non-null   object
 3   patient_givenname  4543 non-null   object
 4   date_of_birth      4543 non-null   object
 5   date_of_death      13 non-null     object
 6   registration_guid  4543 non-null   object
 7   patient_id         4543 non-null   int64 
 8   gender             4543 non-null   object
dtypes: int64(2), object(7)
memory usage: 319.6+ KB


The `.info()` method creates a data description of the dataframe. It tells us:

- How many rows (4543)
- How many columns (9)
- Column names
- How many values are non-null for each column
- The types of data contained within each column
- The size of the dataset loaded into memory (~319.6KB)

Looking at the output of the `.info()` method, we can intepret each column as follows:

We can see that there are 3 different datatypes within this dataset:

- object = this column contains categorical values
- int64 = this column contains integer values (whole numbers)